### PR TITLE
metrics: Add checkmetrics for kata metrics CI

### LINKED ISSUE #7160: add checkmetrics for metrics tests

### PR DESCRIPTION
**Pull request by GabyCT**
This PR adds the checkmetrics scripts that will be used for the kata metrics CI.

Fixes #7160

### CODE REVIEW

File: tests/metrics/cmd/checkmetrics/main.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+Program checkmetrics compares the results from a set of metrics
+results, stored in JSON files, against a set of baseline metrics
+'expectations', defined in a TOML file.
+
+It returns non zero if any of the TOML metrics are not met.
+
+It prints out a tabluated report summary at the end of the run.
```
Comment (dborquez) [inline tests/metrics/cmd/checkmetrics/main.go:12]:
```suggestion
It prints out a tabulated report summary at the end of the run.
```

File: tests/metrics/cmd/checkmetrics/json_test.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+	tmpdir, err := os.MkdirTemp("", "cm-")
+	assert.NoError(err)
+	defer os.RemoveAll(tmpdir)
+
+	// Check a badly formed JSON file
+	badFileName := tmpdir + "badFile.json"
+	err = CreateBadFile(badFileName)
+	assert.NoError(err)
+
+	// Set up our basci metrics struct
```
Comment (dborquez) [inline tests/metrics/cmd/checkmetrics/json_test.go:123]:
```suggestion
	// Set up our basic metrics struct
```

**Review by dborquez — CHANGES_REQUESTED** [review]
thanks @GabyCT, just fix a couple of syntax errors, but lgtm in general, since the number of files in this PR is large, just reviewed the `checkmetrics` files.

**Comment by GabyCT** [comment]
@dborquez changes applied

**Review by dborquez — APPROVED** [review]

**Pull request merged**